beacon-chain/state/testtmpl/getters_validator.go

```diff
@@ -0,0 +1,22 @@
+package testtmpl
+
+import (
+	"testing"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/testing/assert"
+	"github.com/prysmaticlabs/prysm/testing/require"
+)
+
+type testTmplBeaconState_ValidatorAtIndexReadOnly_HandlesNilSlice_factory func() (state.BeaconState, error)
```

Comment:
```suggestion
type getState func() (state.BeaconState, error)
```

Reply:
@terencechain `getState` would be global for the package, and it clash with other factories necessary for other tests templates

Reply:
It's local within the pkg, and it is better given the function signatures. We can make it more explicit if there really is a need (which I don't believe) later